src/Streamly/Internal/Data/Unfold.hs

```diff
@@ -650,6 +673,146 @@ enumerateFromToIntegral to =
 enumerateFromIntegral :: (Monad m, Integral a, Bounded a) => Unfold m a a
 enumerateFromIntegral = enumerateFromToIntegral maxBound
 
+{-# INLINE enumerateFromStepNum #-}
+enumerateFromStepNum :: (Monad m, Num a) => a -> Unfold m a a
+enumerateFromStepNum stride = Unfold step return
+    where
+    {-# INLINE_LATE step #-}
+    step !s = return $ (Yield $! s) $! (s + stride)
+
+{-# INLINE_NORMAL numFrom #-}
+numFrom :: (Monad m, Num a) => Unfold m a a
+numFrom = enumerateFromStepNum 1
+
+{-# INLINE_NORMAL enumerateFromToFractional #-}
+enumerateFromToFractional :: (Monad m, Fractional a, Ord a) => a -> Unfold m a a
+enumerateFromToFractional to =
+    takeWhile (<= to + 1 / 2) $ enumerateFromStepNum 1
```

Comment:
Is there any specific reason to call `takeWhile` which yields till the stream generates elements `<= to + 0.5`?

Reply:
This is more of a copy-paste. Look at the function with the same name in StreamD.hs